### PR TITLE
[DEVEX-104] Setup Yarn

### PR DESCRIPTION
**Pull request by lucacavallaro**
# Use yarn with Plug'n'Play

Date: 2024-03-11

## Status

Accepted

## Context

**Plug'n'Play** is an installation strategy, alternative to the classic resolution algorithm used by `npm`/`node.js` with `node_modules`, that simplify working on monorepo and large javascript codebases.

For our use cases, `pnp` has two major advantages over the classic approach:

1. An efficient dependency resolution algorithm and a global cache that allows dependency reuse (the developer installs dependencies only once on his machine, regardless of how many projects it is used in).

2. Catch _ghost dependencies_ (dependencies that are not listed in `package.json`), ensuring that each workspace declares all dependencies it uses.

See the documentation for more details https://yarnpkg.com/features/pnp

## Decision

We decided to enable `plug'n'play` with global cache on our yarn setup.

### CODE REVIEW

**Comment by gunzip** [comment]
When are we supposed to run `yarn dlx @yarnpkg/sdks` to update them? I'dd add a line into the README.

**Comment by lucacavallaro** [comment]
> When are we supposed to run `yarn dlx @yarnpkg/sdks` to update them? I'dd add a line into the README.

I don't have a clear idea at the moment, I don't even know if it will be necessary. Perhaps the right time is during the periodic update of dependencies.

**Comment by gunzip** [comment]
needs rebasing on main

File: templates/monorepo/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# TypeScript Monorepo Template
```
Comment (balanza) [inline templates/monorepo/README.md:1]: **issue:** I'd add some documentation about it to give developers a little context. Mainly any requirements/configurations, caveats, and little context about files in `.yarn/sdk` folder.
‏

**Review by balanza — CHANGES_REQUESTED** [review]
I think some docs may be useful

**Pull request closed without merging**